web/src/main/java/org/cbioportal/web/util/DataBinHelper.java

```diff
@@ -145,16 +145,17 @@ public Range<BigDecimal> calcInterquartileRange(List<BigDecimal> sortedValues) {
 
     public BigDecimal calcQ1(List<BigDecimal> sortedValues) {
         return sortedValues.isEmpty() ?
-            null : sortedValues.get((int) Math.floor(sortedValues.size() / 4.0));
+            null : sortedValues.get((int) Math.floor(sortedValues.size() * 0.25));
     }
 
     public BigDecimal calcMedian(List<BigDecimal> sortedValues) {
-        return ObjectUtils.median(sortedValues.toArray(new BigDecimal[0]));
+        return sortedValues.isEmpty() ?
+            null : sortedValues.get((int) Math.floor(sortedValues.size() * 0.50));
```

Comment:
What if we have an even number of elements? It doesn't look like a median. I think we should change it, or rename it if we are not calculating `median` at here.
![image](https://user-images.githubusercontent.com/15748980/167653877-d738109a-9eb4-4b24-9b80-3968a50e72e0.png)

Reply:
@dippindots @Luke-Sikina  Well, that existing implementation is wrong! Looking at the implementation of that one, it will cast the data to a series of unigue values (note the **TreeSet**) before evaluating the median:

```
    @SafeVarargs
    public static <T extends Comparable<? super T>> T median(T... items) {
        Validate.notEmpty(items);
        Validate.noNullElements(items);
        TreeSet<T> sort = new TreeSet();
        Collections.addAll(sort, items);
        T result = (Comparable)sort.toArray()[(sort.size() - 1) / 2];
        return result;
    }
```

This is incorrect. The median should take multiple identical observations into account.

Reply:
@dippindots @Luke-Sikina 
>What if we have an even number of elements?
The issue here is that the calcQ1 and calcQ3 functions also do not calculate true first and third quartile boundaries. I did this by extensive testing. The problem here is that there are no tests included for any of these functions. I think you have a valid point here, but I do not think that it is in the scope of this PR.

Reply:
It turns out that the legacy calcQ1 and calcQ3 functions were used to calculate the outer-whiskers of (I presume) a box plot which used some very funky approximations.  I have renamed these functions to valueCloseToQ1 and valueCloseToQ3, respectively, to indicate that these legacy functions do actually calculate quartiles as suggested in their name. I have (re-)implemented the calcQ1, calcMedian, and calcQ3 functions to calculate actual quartile boundaries. These new functions are used for binning of histograms. I have added plenty of tests that should document behavior of all involved functions. Can you re-review is you have time